pkg/test/integration/common/framework.go

```diff
@@ -0,0 +1,772 @@
+		ginkgo.By("Applying kubernetes/crds into control cluster")
+		gomega.Expect(c.ControlCluster.ApplyFiles(filepath.Join(mcmRepoPath, "kubernetes/crds"), controlClusterNamespace)).To(gomega.BeNil())
+
+	}
+
+	// starting controllers
+	if len(mcContainerImageTag) != 0 && len(mcmContainerImageTag) != 0 {
+
+		/* if any of mcmContainerImage  or mcContainerImageTag flag is non-empty then,
+		create/update machinecontrollermanager deployment in the control-cluster with specified image
+		*/
+		if c.ControlCluster.IsSeed(c.TargetCluster) {
+			ginkgo.By("Updating MCM Deployemnt")
+			gomega.Expect(c.prepareMcmDeployment(mcContainerImageTag, mcmContainerImageTag, false)).To(gomega.BeNil())
+		} else {
+			ginkgo.By("Creating MCM Deployemnt")
+			gomega.Expect(c.prepareMcmDeployment(mcContainerImageTag, mcmContainerImageTag, true)).To(gomega.BeNil())
+		}
+
+	} else {
+		/*
+		 run mc and mcm locally
+		*/
+		if c.ControlCluster.IsSeed(c.TargetCluster) {
+			ginkgo.By("Scaledown existing machine controllers")
+			gomega.Expect(c.scaleMcmDeployment(0)).To(gomega.BeNil())
+		}
+
+		ginkgo.By("Starting Machine Controller ")
+		args := strings.Fields(fmt.Sprintf("make --directory=%s start CONTROL_KUBECONFIG=%s TARGET_KUBECONFIG=%s CONTROL_NAMESPACE=%s LEADER_ELECT=false ",
+			"../../..",
+			c.ControlCluster.KubeConfigFilePath,
+			c.TargetCluster.KubeConfigFilePath,
+			controlClusterNamespace))
+		outputFile, err := helpers.RotateLogFile(mcLogFile)
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+		sess, err := gexec.Start(exec.Command(args[0], args[1:]...), outputFile, outputFile)
+		mcsession = sess
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+		gomega.Expect(mcsession.ExitCode()).Should(gomega.Equal(-1))
+
+		ginkgo.By("Starting Machine Controller Manager")
+		args = strings.Fields(fmt.Sprintf("make --directory=%s start CONTROL_KUBECONFIG=%s TARGET_KUBECONFIG=%s CONTROL_NAMESPACE=%s LEADER_ELECT=false MACHINE_SAFETY_OVERSHOOTING_PERIOD=300ms",
+			mcmRepoPath,
+			c.ControlCluster.KubeConfigFilePath,
+			c.TargetCluster.KubeConfigFilePath,
+			controlClusterNamespace))
+		outputFile, err = helpers.RotateLogFile(mcmLogFile)
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+		mcmsession, err = gexec.Start(exec.Command(args[0], args[1:]...), outputFile, outputFile)
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+		gomega.Expect(mcmsession.ExitCode()).Should(gomega.Equal(-1))
+	}
+
+	ginkgo.By("Setup MachineClass")
+	gomega.Expect(c.setupMachineClass()).To(gomega.BeNil())
+
+	// initialize orphan resource tracker
+	ginkgo.By("Looking for machineclass resource in the control cluster")
+	machineClass, err := c.ControlCluster.McmClient.MachineV1alpha1().MachineClasses(controlClusterNamespace).Get(testMachineClassResources[0], metav1.GetOptions{})
+	if err == nil {
+		secret, err := c.ControlCluster.Clientset.CoreV1().Secrets(machineClass.SecretRef.Namespace).Get(machineClass.SecretRef.Name, metav1.GetOptions{})
```

Comment:
With this [PR](https://github.com/gardener/machine-controller-manager/pull/578) on MCM, we have two secrets introduced - `credentialsSecretRef` and the old `secretRef` in the MachineClass. Since, in Gardener ecosystem, mostly all MachineClasses use the same credentials, the actual cloud credentials are stored in the secret referred by `credentialsSecretref` in the MachineClass while the userData exists in the secret referred by `secretRef`.

Currently, with this integration test suite, the credentials are attempted to be fetched from the secrets under `secretRef` field of machineClass which may lead to error like below:

```
STEP: Initializing orphan resource tracker
Failure [10.694 seconds]
[BeforeSuite] BeforeSuite
/Users/i538135/go/src/github.com/gardener/machine-controller-manager-provider-aws/test/integration/controller/controller_test.go:41

  Unexpected error:
      <*status.Status | 0xc000165ae8>: {
          code: 13,
          message: "Error while validating ProviderSpec [secretRef.AWSAccessKeyID: Required value: Mention atleast providerAccessKeyId or accessKeyID, secretRef.AWSSecretAccessKey: Required value: Mention atleast providerSecretAccessKey or secretAccessKey]",
      }
      machine codes error: code = [Internal] message = [Error while validating ProviderSpec [secretRef.AWSAccessKeyID: Required value: Mention atleast providerAccessKeyId or accessKeyID, secretRef.AWSSecretAccessKey: Required value: Mention atleast providerSecretAccessKey or secretAccessKey]]
  occurred

  /Users/i538135/go/src/github.com/gardener/machine-controller-manager-provider-aws/vendor/github.com/gardener/machine-controller-manager/pkg/test/integration/common/framework.go:452
```

So, I would recommend, fetching the credentials from `secretRef` and if not found, then fetch it from `credentialsSecretRef` field of MachineClass.

Reply:
Thanks for the finding. I have made some changes and should work fine now.